src/common/signalk/SKNMEA2000Parser.cpp

```diff
@@ -99,23 +149,155 @@ const SKUpdate& SKNMEA2000Parser::parse128267(const SKSourceInput& input, const
 
     update->setEnvironmentDepthBelowTransducer(depthBelowTransducer);
 
-    if (!N2kIsNA(offset)) {
-      // When offset is negative, it's the distance between transducer and keel
-      if (offset < 0) {
-        update->setEnvironmentDepthTransducerToKeel(offset * -1);
-        update->setEnvironmentDepthBelowKeel(depthBelowTransducer + offset);
-      }
-      else if (offset > 0) {
-        update->setEnvironmentDepthSurfaceToTransducer(offset);
-        update->setEnvironmentDepthBelowSurface(depthBelowTransducer + offset);
+    if (N2kIsNA(offset)) {
+      // When no offset then offset should be Zero
+      offset = 0;
+    }
+    // When offset is negative, it's the distance between transducer and keel
+    if (offset < 0) {
+      update->setEnvironmentDepthTransducerToKeel(offset * -1);
+      update->setEnvironmentDepthBelowKeel(depthBelowTransducer + offset);
+    }
+    else if (offset > 0) {
+      update->setEnvironmentDepthSurfaceToTransducer(offset);
+      update->setEnvironmentDepthBelowSurface(depthBelowTransducer + offset);
+    }
+
+    _sku = update;
+    return *_sku;
+  }
+  else {
+    DEBUG("Unable to parse N2kMsg with PGN %i", msg.PGN);
+    return _invalidSku;
+  }
+}
+
+// *****************************************************************************
+//   PGN 130306 W I N D
+//    tN2kWindReference:
+//      N2kWind_True_North=0,
+//      N2kWind_Magnetic=1,
+//      N2kWind_Apprent=2,
+//      N2kWind_True_boat=3
+// *****************************************************************************
+const SKUpdate& SKNMEA2000Parser::parse130306(const SKSourceInput& input, const tN2kMsg& msg, const SKTime& timestamp) {
+  unsigned char sid;
+  double windSpeed;     // Knots
+  double windAngle;     // in Radians 0... 2*pi
+  tN2kWindReference windReference;
+
+  if (ParseN2kPGN130306(msg,sid,windSpeed,windAngle,windReference) ) {
+    SKUpdateStatic<2> *update = new SKUpdateStatic<2>();
+    update->setTimestamp(timestamp);
+
+    SKSource source = SKSource::sourceForNMEA2000(input, msg.PGN, msg.Priority, msg.Source);
+    update->setSource(source);
+
+    if (!N2kIsNA( windAngle) && !N2kIsNA( windSpeed)) {
+      if (windReference == N2kWind_Apprent) {
+        // -> m/s (Meters per second)
+        update->setEnvironmentWindSpeedApparent(SKKnotToMs(windSpeed));
+        DEBUG("AWS: %d kts",windSpeed);
+        // -> Apparent wind angle, rad, negative to port
+        if (windAngle >M_PI) windAngle *= -1;
+        update->setEnvironmentWindAngleApparent(windAngle);
+        DEBUG("AWA: %d °",SKRadToDeg(windSpeed));
       }
     }
+    // We do not take True Wind Speed as we do not know how the system has calculated it!!
```

Comment:
I think we do. Look at the different options: https://github.com/ttlappalainen/NMEA2000/blob/master/src/N2kMessages.h#L150

And then we can also take inspiration from [what n2k-signalk does with this PGN](https://github.com/SignalK/n2k-signalk/blob/master/pgns/130306.js):
- Apparent is obvious => `environment.wind.speedApparent` and `environment.wind.angleApparent`.
- True boat is true calculated to the boat speed => `environment.wind.speedTrue` and `environment.wind.angleTrueWater`
- True water is true calculated to the SOG => `environment.wind.speedOverGround` and `environment.wind.directionTrue`
- The last two ones are harder. `n2k-signalk` does not support them. I would guess that they are both apparent but using the boat heading to calculate a wind direction instead of a wind angle. With true north, the angle has just been corrected for the declination. => `environment.wind.speedApparent` and `environment.wind. directionTrue` or `.directionMagnetic` (I am not sure about this one).

@ttlappalainen do you have any tips for us here?

I think the best thing to do here would be to collect some samples from boats (do you have a nmea2000 sensor?) and add some tests to make sure we are doing it right.

You have found the unit tests but did you look [at the integration tests](https://github.com/sarfata/kbox-firmware/blob/develop/src/sktool/tests/n2k-transducer.json)? I have not documented this at all but it's fun to try! Travis runs them automatically. On Linux/Mac you can try with just `make validation` or just run `src/sktool/sktool-test.py --implementation sktool src/sktool/tests/*json` to run them locally.

Reply:
I should have written "True Wind will be done later..." :smiley:

1. I think it will be of importance, that a user can choose if she wants TrueWind from the bus or calculated in KBox 

2. True Boat makes sense as it should be the "True (Sailing) Wind" referred to water

3. N2kWind_True_North=0 ->  is true calculated to the SOG => environment.wind.speedOverGround and environment.wind.directionTrue 

=> I will follow up with a translation of 2. and 3.

4. N2kWind_Magnetic=1 => further investigations needed

To your Questions:
I built a simple data-logger with a modified KBox-program, but had no chance to be on a boat yet.

It would be nice to have more of different logfiles @ttlappalainen mentioned once that he has a lot of them.
My only source at the moment is:
Kees Merrimac file: https://github.com/canboat/canboat/blob/master/samples/merrimac-actisense-serial-2011.raw
(I just saw that there are some new files there, I will have to look at)

More of those files would give us a nice opportunity to do some "live"-tests

I found the other test files, but could not sort it out, how to handle them. 
Thank you for your description, looking forward to try it!